contracts/orbital-core/src/contract.rs

```diff
@@ -1,119 +1,171 @@
     }
+}
 
-    // save an empty user config
-    USER_CONFIGS.save(deps.storage, info.sender, &UserConfig::default())?;
+fn query_user_addresses(deps: QueryDeps) -> StdResult<Vec<String>> {
+    let users = USER_CONFIGS.range(deps.storage, None, None, cosmwasm_std::Order::Ascending);
 
-    Ok(Response::new().add_attribute("method", "register_user"))
-}
+    let mut user_addrs = vec![];
+    for user_entry in users {
+        user_addrs.push(user_entry?.0);
+    }
 
-fn admin_update_ownership(
-    deps: DepsMut,
-    block: &BlockInfo,
-    sender: &Addr,
-    action: Action,
-) -> Result<Response, ContractError> {
-    let resp = update_ownership(deps, block, sender, action)?;
-    Ok(Response::default().add_attributes(resp.into_attributes()))
+    Ok(user_addrs)
 }
 
-fn admin_register_new_domain(
-    deps: DepsMut,
-    info: MessageInfo,
+fn query_clearing_account(
+    deps: QueryDeps,
     domain: String,
-    account_type: UncheckedOrbitalDomainConfig,
-) -> Result<Response, ContractError> {
-    // only the owner can register new domains
-    assert_owner(deps.storage, &info.sender)?;
-
-    // validate the domain configuration
-    let orbital_domain = account_type.try_into_checked(deps.api)?;
-
-    // ensure the domain does not already exist
-    if ORBITAL_DOMAINS.has(deps.storage, domain.to_string()) {
-        return Err(ContractError::OrbitalDomainAlreadyExists(
-            domain.to_string(),
-        ));
-    }
+    addr: String,
+) -> StdResult<Option<String>> {
+    let user_config = USER_CONFIGS.load(deps.storage, addr)?;
+    let ica_id = get_ica_identifier(user_config.id, domain);
+    CLEARING_ACCOUNTS.load(deps.storage, ica_id)
+}
 
-    // store the validated domain config in state
-    ORBITAL_DOMAINS.save(deps.storage, domain.to_string(), &orbital_domain)?;
+fn query_ownership(deps: QueryDeps) -> StdResult<cw_ownable::Ownership<Addr>> {
+    get_ownership(deps.storage)
+}
+
+fn query_orbital_domain(deps: QueryDeps, domain: String) -> StdResult<OrbitalDomainConfig> {
+    ORBITAL_DOMAINS.load(deps.storage, domain)
+}
 
-    Ok(Response::default()
-        .add_attribute("method", "register_new_domain")
-        .add_attribute("domain", domain))
+fn query_user_config(deps: QueryDeps, user: String) -> StdResult<UserConfig> {
+    USER_CONFIGS.load(deps.storage, user)
 }
 
 #[entry_point]
-pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
-    match msg {
-        QueryMsg::Ownership {} => to_json_binary(&get_ownership(deps.storage)?),
-        QueryMsg::OrbitalDomain { domain } => query_orbital_domain(deps, domain),
-        QueryMsg::UserConfig { user } => query_user_config(deps, user),
-    }
+pub fn reply(_deps: ExecuteDeps, _env: Env, _msg: Reply) -> StdResult<Response<NeutronMsg>> {
+    Ok(Response::default())
+}
+
+#[entry_point]
+pub fn migrate(_deps: ExecuteDeps, _env: Env, _msg: MigrateMsg) -> StdResult<Response<NeutronMsg>> {
+    Ok(Response::default())
```

Comment:
nit: I'd remove them if you are not using them

Reply:
they will be used soon. just added them in advance for the plumbing done in testing suites.

Reply:
updated with `unimplemented!()` instead of an `Ok` response & made an issue for migration